### ISSUE TITLE
HCF: "Mode Random" randomily fails to find a hcf 5

### ISSUE THREAD

**Issue by McBen** [mike40033/iitc-57Cell#11]
example portals:
https://intel.ingress.com/?pll=51.392946,6.759832
https://intel.ingress.com/?pll=51.392109,6.7721
https://intel.ingress.com/?pll=51.397653,6.768505

with these anchors you can build a HFC Level 5
While mode `perfect` always finds a solution
mode `random` fails sometimes (randomly, roughly every second time)

Failed (with example solution)
![grafik](https://github.com/mike40033/iitc-57Cell/assets/29117/1e79c221-7c89-41a5-815f-57141270d5bc)
found one (with an similar solution)
![grafik](https://github.com/mike40033/iitc-57Cell/assets/29117/b54c7317-eada-4429-97ef-009f59041127)

**Comment by mike40033** [comment]
Notes:

* Brunnen, Arm in Arm and Gruft consistently fails to yield a level 3 HCF, 
* Stone Frau, Arm in Arm and Gruft consistently succeeds.

One of these is wrong. 

I suspect it's to do with whether Marienbild 5 is north or south of the line from Gruft to Arm in Arm, and that portalInTriangle is giving inconsistent answers. 

@McBen I'd appreciate a field test to see whether northward or southward links from Marienbild 5 are blocked when Gruft and Arm in Arm are linked.

Portal links: 

Gruft: https://intel.ingress.com/intel?ll=51.397653,6.768505&z=17&pll=51.397653,6.768505
Arm in Arm: https://intel.ingress.com/intel?ll=51.395171,6.766774&z=17&pll=51.395171,6.766774
Marienbild 5: https://intel.ingress.com/intel?ll=51.396221,6.767506&z=17&pll=51.396221,6.767506
Brunnen: https://intel.ingress.com/intel?ll=51.394453,6.763827&z=17&pll=51.394453,6.763827
Stone Frau: https://intel.ingress.com/intel?ll=51.395297,6.765352&z=17&pll=51.395297,6.765352

~ 57Cell

**Comment by McBen** [comment]
"Portal Marienbild is RIGHT of a link from GRUFT to ARM_IN_ARM" 

used code/math:
https://jsfiddle.net/mcben/j6enxyar

**Comment by McBen** [comment]
FYI: I'm usually using a different `PointInTriangle` approach.

I can't say if this one is better or worse - atleast it doesn't require a tolerance value (`eps`)
```typescript
export const isPortalInField = (p: L.LatLng, field: [L.LatLng, L.LatLng, L.LatLng]): boolean => {

    const ct1 =  convertTo3D(field[0]);  
    const ct2 =  convertTo3D(field[1]);
    const ct3 =  convertTo3D(field[2]);
    const cp =  convertTo3D(p);

    const np1 = cross(ct1, ct2);
    const np2 = cross(ct2, ct3);
    const np3 = cross(ct3, ct1);

    const direction = dot(np1, ct3);

    const d1 = dot(np1, cp) * direction;
    const d2 = dot(np2, cp) * direction;
    const d3 = dot(np3, cp) * direction;

    return (d1 > 0 && d2 > 0 && d3 > 0);
}
```
The idea here: the point must be on the same side of every field link (greatcircle-plane). `direction` is used to test if field is portals are clockwise (to skip another "if" case).

**Comment by McBen** [comment]
Looks like the problem is here:
https://github.com/mike40033/iitc-57Cell/blob/master/plugins/homogeneous-fields/homogeneous-fields.user.js#L301

I can't name the situation but seems like a sub Portal is a valid center while current center is not.

**Comment by mike40033** [comment]
Fixed in version 1.2.2

Thanks!
